### DISCUSSION TITLE
Environment variables are not getting picked up after deployment

### DISCUSSION THREAD

**Discussion by Codeskb** [microsoft/sample-app-aoai-chatGPT#914] [Q&A]
Hello team,
I used below az commands to deploy my local code to azure.
```
az webapp up --runtime <runtime-stack> --sku <sku> --name <existing-app-name> --resource-group <resource-group-name>
az webapp config set --startup-file "python3 -m gunicorn app:app" --name <existing-app-name>
```
I have .env file with environment variables and works fine locally, but the deployment does not work and throws 
```
Exception("AZURE_OPENAI_ENDPOINT or AZURE_OPENAI_RESOURCE is required")
ERROR:root:Exception in send_chat_request
```
How to I set env variables in azure ?

Thanks

**Comment by stbere** [comment]
try running `az provision` first, see if that fixes it for you.

**Reply by Darshan-Dholam** [reply]
have you fixed it ?

**Reply by Codeskb** [reply]
As I have already deployed the solution I did not run az provision, I have used in the source code directly.
